src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets

```diff
@@ -45,6 +45,43 @@ Copyright (c) .NET Foundation. All rights reserved.
     <_GeneratedStaticWebAssetsDevelopmentManifest>$(IntermediateOutputPath)Microsoft.AspNetCore.StaticWebAssets.xml</_GeneratedStaticWebAssetsDevelopmentManifest>
   </PropertyGroup>
 
+  <!-- 
+    Targets used for publishing content from razor class libraries
+  -->
+
+  <!--
+    Called after ComputeFilesToPublish but before CopyFilesToPublishDirectory - this target is called when 
+    publishing the project to get a list of files to the output directory.
+  -->
+  <Target
+    Name="_StaticWebAssetsComputeFilesToPublish"
+    AfterTargets="ComputeRefAssembliesToPublish"
+    DependsOnTargets="ResolveStaticWebAssetsInputs">
+
+    <ItemGroup>
+      
+      <!-- Filter the static web assets foreign to the project and then add them to the list of resolved
+           files to publish.
+       -->
+      <_ExternalPublishStaticWebAsset 
+        Include="%(StaticWebAsset.FullPath)"
+        Condition="'%(SourceType)' != ''">
+        
+        <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+        <RelativePath>wwwroot\%(BasePath)\%(RelativePath)</RelativePath>
+        <TargetPath>wwwroot\%(BasePath)\%(RelativePath)</TargetPath>
+      
+      </_ExternalPublishStaticWebAsset>
```

Comment:
Is there a reason why we need this itemgroup instead of adding to `ResolvedFileToPublish` directly? Just for logging?

Reply:
Yes, MSbuild complains about missing metadata when batching if I try to do it directly on ResolvedFileToPublish, because I think the metadata on the itemgroups needs to match when batching.